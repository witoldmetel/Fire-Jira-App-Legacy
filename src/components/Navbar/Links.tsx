import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { Dropdown } from 'semantic-ui-react';

import { Auth, NewUser } from '../../fixtures/types';
import { ACCOUNT_TYPE } from '../../fixtures/constants';
import { signOut } from '../../store/actions';

export interface Props {
  projectId: string;
  profile: NewUser;
  auth: Auth;
  signOut: () => void;
}

class Links extends React.Component<Props> {
  private get createProjectLink() {
    const { profile } = this.props;

    return profile.accountType === ACCOUNT_TYPE.VIP && !this.props.projectId ? (
      <NavLink to="/project/new" className="header item">
        Create New Project
      </NavLink>
    ) : null;
  }

  private get createTaskLink() {
    return this.props.projectId ? (
      <NavLink to={`/project/${this.props.projectId}/task/new`} className="header item">
        Create New Task
      </NavLink>
    ) : null;
  }

  private get renderLinks() {
    const { auth, signOut } = this.props;

    return auth.uid ? (
      <React.Fragment>
        {this.createProjectLink}
        {this.createTaskLink}
        <Dropdown
          item
          icon={<img src={`https://api.adorable.io/avatars/${auth.uid}.png`} className="ui mini circular image" />}
          simple
        >
          <Dropdown.Menu direction="left">
            <Dropdown.Item disabled>Settings</Dropdown.Item>
            <Dropdown.Item onClick={() => signOut()}>Logout</Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </React.Fragment>
    ) : (
      <React.Fragment>
        <NavLink to="/signup" className="header item">
          Register
        </NavLink>
        <NavLink to="/signin" className="header item">
          Login
        </NavLink>
      </React.Fragment>
    );
  }

  public render() {
    return <div className="right menu">{this.renderLinks}</div>;
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    auth: state.firebase.auth,
    profile: state.firebase.profile,
    projectId: ownProps.match.params.id
  };
};

export default connect(mapStateToProps, { signOut })(Links);
