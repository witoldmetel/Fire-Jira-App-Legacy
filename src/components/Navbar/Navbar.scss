.ui.menu .ui.container {
  width: 100%;
  margin: 0 !important; // Unfortunately needed to replace semantic ui styles

  .header.item .logo {
    background-image: url('../../../public/assets/logo.png');
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    width: 40px;
    height: 40px;
    margin-right: 10px;
  }
}
