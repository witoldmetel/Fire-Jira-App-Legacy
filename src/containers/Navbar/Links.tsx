import React from 'react';
import { NavLink } from 'react-router-dom';

export default class Links extends React.Component {
  private get renderLinks() {
    // todo: Add later proper state
    return true ? (
      <React.Fragment>
        <NavLink to="/z" className="header item">
          Create New Task
        </NavLink>
        <NavLink to="/x" className="header item">
          Logout
        </NavLink>
        <NavLink to="/x" className="header item">
          <img src="https://api.adorable.io/avatars/profile.png" className="ui mini circular image" />
        </NavLink>
      </React.Fragment>
    ) : (
      <React.Fragment>
        <NavLink to="/x" className="header item">
          Signup
        </NavLink>
        <NavLink to="/s" className="header item">
          Login
        </NavLink>
      </React.Fragment>
    );
  }

  public render() {
    return <div className="right menu">{this.renderLinks}</div>;
  }
}
