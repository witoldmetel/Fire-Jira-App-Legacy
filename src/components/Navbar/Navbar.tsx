import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import classnames from 'classnames';

import Links from './Links';
import { Auth } from '../../fixtures/types';

import './Navbar.scss';

export interface Props {
  auth: Auth;
}

class Navbar extends Component<Props> {
  private get className() {
    return classnames('ui menu', {
      main: !this.props.auth.uid
    });
  }

  public render() {
    console.log(this.props);
    return (
      <header className={this.className}>
        <nav className="ui container">
          <Link to="/" className="header item">
            <span className="logo" />
            <p className="brand-name">Fire Jira</p>
          </Link>
          <Links {...this.props} />
        </nav>
      </header>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    auth: state.firebase.auth
  };
};

export default connect(mapStateToProps)(Navbar);
