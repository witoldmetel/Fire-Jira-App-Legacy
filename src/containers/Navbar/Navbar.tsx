import React from 'react';
import { Link } from 'react-router-dom';

import Links from './Links';

class Navbar extends React.Component {
  public render() {
    return (
      <header className="ui menu">
        <nav className="ui container">
          <Link to="/" className="header item">
            <img className="logo" src={`https://api.adorable.io/avatars/${Math.random()}.png`} />
            Task List App
          </Link>
          <Links />
        </nav>
      </header>
    );
  }
}

export default Navbar;
