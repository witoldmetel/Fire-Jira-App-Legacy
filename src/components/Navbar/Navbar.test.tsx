import * as React from 'react';
import { shallow } from 'enzyme';

import Navbar from './Navbar';

describe.skip('<Navbar>', () => {
  it('renders without crashing', () => {
    shallow(<Navbar />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Navbar />);

    expect(wrapper).toMatchSnapshot();
  });
});
