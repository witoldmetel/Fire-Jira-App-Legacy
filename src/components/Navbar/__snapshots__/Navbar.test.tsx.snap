// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar> renders correctly 1`] = `
<header
  className="ui menu"
>
  <nav
    className="ui container"
  >
    <Link
      className="header item"
      to="/"
    >
      <span
        className="logo"
      />
      <p>
        Fire Jira
      </p>
    </Link>
    <Connect(Links) />
  </nav>
</header>
`;
